### ISSUE TITLE
Markdown not working in custom components

### ISSUE THREAD

**Issue by vanbroup** [fazouane-marouane/remark-jsx#15]
@fazouane-marouane after the fix from #10 it's possible to use headers inside a custom component, using a list, links or images doesn't seem to work though.

A list is not rendered:
```html
<Component>
- first
- second 
- third
</Component>
```

```html
<Component>
* first
* second 
* third
</Component>
```

```html
<Component>
+ first
+ second 
+ third
</Component>
```

Links or images break the rendring process completly:

```html
<Component>
[I'm an inline-style link](https://www.google.com)
</Component>
```

```html
<Component>
![alt text](http://path/to/img.jpg "Title")
</Component>
```

Less imported (for me) but confirmed not working:
```html
<Component>
| Tables        | Are           | Cool  |
| ------------- |:-------------:| -----:|
| col 3 is      | right-aligned | $1600 |
| col 2 is      | centered      |   $12 |
| zebra stripes | are neat      |    $1 |
</Component>
```

```html
<Component>
Should get a horizontal rule below
---
And bove this text
</Component>
```

```html
<Component>
> This is blockquoted text.
</Component>
```

**Comment by fazouane-marouane** [comment]
Hi @vanbroup 

Actually all the examples you provided already work with the fix I pushed yesterday on npm.
Here is for example the result of the last code snippet: https://runkit.com/fazouane-marouane/5a3d5961e0b950001106609f

It gives:
```json
{
  "type": "root",
  "children": [
    {
      "type": "element",
      "tagName": "Component",
      "properties": {},
      "children": [
        {
          "type": "element",
          "tagName": "blockquote",
          "properties": {},
          "children": [
            {
              "type": "paragraph",
              "children": [
                {
                  "type": "text",
                  "value": "This is blockquoted text."
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```

I think that somehow you don't have the latest version (0.5.6) in your node_modules.

**Comment by creativegeniusod** [comment]
Hi @fazouane-marouane 

I have create a Center Component. Below is code used in Center component.

```
render() {
    const {
      children
    } = this.props

    return (
      <div>
        <Container>
          <Segment padded basic style={{textAlign:'center'}}>
            {children}
          </Segment>
        </Container>
      </div>
    );
  }
```

And below script is added in markdown file.

```
<Center>
# This is heading tag.
> This is blockquoted text

- first
- second
- third

</Center>

```
It gives me  this output.

``` 
{
  "type": "root",
  "children": [
    {},
    {
      "type": "element",
      "tagName": "Center",
      "properties": {},
      "children": [
        {
          "type": "element",
          "tagName": "h1",
          "properties": {},
          "children": [
            {
              "type": "text",
              "value": "This is heading tag."
            }
          ]
        },
        {
          "type": "element",
          "tagName": "blockquote",
          "properties": {},
          "children": [
            {
              "type": "paragraph",
              "children": [
                {
                  "type": "text",
                  "value": "This is blockquoted text"
                }
              ]
            }
          ]
        },
        {
          "type": "element",
          "tagName": "ul",
          "properties": {},
          "children": [
            {
              "type": "listItem",
              "loose": false,
              "checked": null,
              "children": [
                {
                  "type": "paragraph",
                  "children": [
                    {
                      "type": "text",
                      "value": "first"
                    }
                  ]
                }
              ]
            },
            {
              "type": "listItem",
              "loose": false,
              "checked": null,
              "children": [
                {
                  "type": "paragraph",
                  "children": [
                    {
                      "type": "text",
                      "value": "second"
                    }
                  ]
                }
              ]
            },
            {
              "type": "listItem",
              "loose": false,
              "checked": null,
              "children": [
                {
                  "type": "paragraph",
                  "children": [
                    {
                      "type": "text",
                      "value": "third"
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}

```
But in frontend its show only h1 tag text. Blockquoted and list text not showing. Please check attach screenshot for html output.
![34347581 89f3f3a4 ea2a 11e7 99b9 f9c5151de800 png png image 1920 x 1080 pixels](https://user-images.githubusercontent.com/22888194/34352884-1d5f5f78-ea4b-11e7-89eb-b2d27d616c96.png)


https://github.com/Skoli-Code/skoli-app-template I am follow this example app for using React components in Markdown source.

**Comment by fazouane-marouane** [comment]
@creativegeniusod I'll look into this by the end of the week. It may be unrelated to your issue, but it seems that we shouldn't have/need paragraphs inside list items. So I'll look into that too.

**Comment by vanbroup** [comment]
@fazouane-marouane did you found a possible cause of the problem?

**Comment by fazouane-marouane** [comment]
@vanbroup Sorry I haven't had time to push a fix. It turned out both issues are related; some parts of the AST aren't compiled to html AST. I have some time available this weekend. I'll try to push a fix by then.

**Comment by fazouane-marouane** [comment]
Hello folks!

Sorry for the (very) long wait. I finally got some time to spend on the fix. It's pushed on npm (v0.5.7) 😊
Don't hesitate to share your feedback or report further issues.

**Comment by fazouane-marouane** [comment]
Closing this as the last bugs have been fixed (as of v0.6.0).
Feel free to create an issue if you stumble upon a new limitation.
